Add `seq-of`, `all-of`, and `any-of` combinators (#191)
nubank/matcher-combinators: CHANGELOG.md

[keepachangelog.com](http://keepachangelog.com/).

## Unreleased

> - Introduce `seq-of`, `any-of`, and `all-of` matchers combinators
>    * `seq-of` takes an expected matcher and creates a new matcher over a
>      sequence, where each element matches the provided expected matcher.
>    * `any-of` takes any number of matchers, successfully matching when at least
>      one matches.
>    * `all-of` takes any number of matchers, successfully matching when all
>      match.
> 

Reviewer: ```suggestion
- Add `seq-of` matcher, which takes a matcher, successfully matching when each element matches the provided matcher.
* Add `any-of` matcher, which takes any number of matchers, successfully matching when at least one matches.
* Add `all-of` matcher, which takes any number of matchers, successfully matching when all match.

```
